evm/next/documentation/migrations/add-evm-to-existing-chain.mdx

```diff
@@ -0,0 +1,199 @@
+          "display": "token",
+          "name": "Token",
+          "symbol": "TKN",
+          "denom_units": [
+            {
+              "denom": "atoken",
+              "exponent": 0
+            },
+            {
+              "denom": "token",
+              "exponent": 18
+            }
+          ]
+        }
+      ]
+    }
+  }
+}
+```
+
+### Option B: Non-18-Decimal Token with PreciseBank
+
+For chains with non-18-decimal gas tokens (e.g., 6 decimals like `uatom`), use the [`x/precisebank`](https://github.com/cosmos/evm/tree/main/x/precisebank) module to track fractional balances at the EVM level while maintaining the native denomination for Cosmos transactions.
+
+The `precisebank` module wraps the standard `bank` module and stores fractional balances separately. For example, if someone transfers 0.5 × 10^-6 tokens (less than 1 `uatom`) via an EVM transaction, precisebank tracks this fractional amount even though the base denomination can't represent it.
+
+### Decimal Configuration Summary
+
+| Token Type | Configuration | Module Required |
+|------------|---------------|-----------------|
+| 18-decimal native token | Standard setup | None |
+| Non-18-decimal token | ExtendedDenomOptions + precisebank | `x/precisebank` |
+
+---
+
+## 3) Upgrade Handler Implementation
+
+After configuring address derivation and gas token decimals, implement an upgrade handler to add the EVM module to your running chain.
+
+<Info>
+Refer to the [upgrade handlers documentation](./upgrade-handlers) for more details on implementing and testing upgrade handlers.
+</Info>
+
+---
+
+## 4) Testing and Validation
+
+Before proposing the upgrade on mainnet, thoroughly test on a testnet or local network.
+
+### Pre-Upgrade Checklist
+
+- Coin type set to 60
+- EthSecp256k1 added to keyring and signing options
+- Gas token decimal configuration determined (18-decimal native or precisebank)
+- Upgrade handler implemented and tested
+
+### Post-Upgrade Validation
+
+After the upgrade executes:
+
+1. **Verify address derivation:**
+   ```bash
+   # Create a new key with eth_secp256k1
+   <binary> keys add test-key --keyring-backend test --algo eth_secp256k1
+
+   # Verify the address format matches Ethereum expectations
+   ```
+
+2. **Test EVM transactions:**
+   ```bash
+   # Send a simple EVM transaction
+   # Verify it processes correctly and gas is calculated in wei
+   ```
+
+3. **Check module parameters:**
+   ```bash
+   <binary> query vm params
+   <binary> query feemarket params
+   ```
+---
+
+## Key Considerations
+
+### Chain Halt Risk
+
+<Warning>
+Improper configuration can cause consensus failures. Always test the complete upgrade process on a testnet that mirrors your mainnet state before proposing the governance upgrade.
+</Warning>
+
+### Account Migration
+
+Existing accounts on your chain will continue to work after adding the EVM module. However:
+
+- Legacy accounts using `secp256k1` keys can still interact with Cosmos SDK modules
```

Comment:
again, point out that a completely different address will be generated with the same priv key/mnemonic if they try to use it on metamask, and they no balances or state would show up